### PR TITLE
fix: Fix Storybook crashes caused by circular object references on uncalled Icon components

### PR DESCRIPTION
**Pull request by codecaaron**
#### Storybook Behaviors
1. Storybook does some [prettifying of JSX](https://github.com/storybookjs/storybook/blob/next/addons/docs/src/frameworks/react/jsxDecorator.tsx#L117 ) which uses a [recursive object property sort method](https://github.com/algolia/react-element-to-jsx-string/blob/master/src/formatter/sortObject.js#L3). 
2. Args are stringified for presentation in various places.

#### Emotion Behavior 
Uncalled Emotion component functions store name spaced object references that are circular on each emotion instance (`__emotion_real`) which is infinitely deep.  It is unclear what the purpose of this is.

#### Cause of Bug
After introducing system props to Icons - stories that consumed Icon functions as a prop broke due to infinite loops attempting to serialize the function.  

#### Fix
This fix for this is to avoid having icons register as an emotion component and stay a `React.forwardRef` while still having access to emotion styles.  The ideal approach would be to fully replace the `svg` tag with a custom `Svg`  emotion component internal to the forwardRef.  However this is not possible as there are limited ways to customize the actual JSX of `svgr` without a custom babel plugin.

This solution uses the render prop `ClassNames` function exposed by emotion/react to generate a the styles prior and apply them as a classname directly to the intrinsic svg component.  As we no longer have access to the theme context default to all styled components we also need to add a `useTheme()` hook to each Icon to ensure that the system props will not break.

```tsx
<ClassNames>{({ css , cx }) => <svg className={cx(className, css(systemProps(props))} />}</ClassNames>
```

This should have no downstream affects to behavior.

#### Other changes
- Upgraded storybook dependencies just in case to the latest minor version (likely the last before 6.3).

### PR Checklist

- [ ] Related to designs:
- [x] Related to JIRA ticket: ABC-123
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60ad6963ffb6b900ec2ff68b--gamut-preview.netlify.app>https://60ad6963ffb6b900ec2ff68b--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60ad6963ffb6b900ec2ff68b>Deploy Logs</a></p>

File: packages/gamut-icons/icon-template.js [hunk 1]
```diff
@@ -1,31 +1,37 @@
...
       size,
       color,
       width,
       height,
+      className: cn,
       ...rest
     },
       svgRef
     ) => {
+      const theme = useTheme();
```
Comment (JoshuaKGoldberg) [inline packages/gamut-icons/icon-template.js:27]: Is this going to cause unit test breakages in the monolith for icons that aren't yet wrapped in gamut-tests?

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@4.14.3-alpha.170481.0<br/>@codecademy/gamut-kit@0.3.118-alpha.170481.0<br/>@codecademy/gamut-labs@18.2.1-alpha.170481.0<br/>@codecademy/gamut@33.1.4-alpha.170481.0<br/>@codecademy/styleguide@40.0.6-alpha.170481.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60ae634252267c0fc993ac3f--gamut-preview.netlify.app>https://60ae634252267c0fc993ac3f--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60ae634252267c0fc993ac3f>Deploy Logs</a></p>

**Review by JoshuaKGoldberg — APPROVED** [review]
Confirmed this fixes the pages, thank you!

We should probably finally get around to writing _runtime_ tests for Storybook pages...

File: packages/gamut-icons/icon-template.js [hunk 2]
```diff
@@ -1,31 +1,37 @@
 const path = require('path');
-const { startCase } = require('lodash');
+const { startCase } = require('lodash/fp');
```
[the remark this answers is not in the document]
Comment (codecaaron) [reply packages/gamut-icons/icon-template.js:2]: Relic from previous attempt to edit the returned JSX ast Opening and closing tags immutably which worked but  ended up causing SVGR to recognize anything... Will update!

File: packages/gamut-icons/icon-template.js (hunk 1 above)
Comment (codecaaron) [reply packages/gamut-icons/icon-template.js:27]: This returns undefined if the context isn't available so the behavior will be the same as `withComponent` or `styled()`, not any better buuut not any worse.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@4.14.3-alpha.a1e4b3.0<br/>@codecademy/gamut-kit@0.3.119-alpha.a1e4b3.0<br/>@codecademy/gamut-labs@18.2.2-alpha.a1e4b3.0<br/>@codecademy/gamut@33.1.5-alpha.a1e4b3.0<br/>@codecademy/styleguide@40.0.7-alpha.a1e4b3.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60ae731bdfc1c727f797f150--gamut-preview.netlify.app>https://60ae731bdfc1c727f797f150--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60ae731bdfc1c727f797f150>Deploy Logs</a></p>

**Pull request merged**